fpdcc/ccfp-asset-dashboard
# Feature/phase and funding form

## Overview

This allows for the manipulation of multiple funding streams from a phase edit page. Features include:
- adding multiple new fundings at once
- editing multiple existing fundings at once
- warning the user if new fundings have not been saved when navigating away
- front and back end validation of values

Closes #156 

### Demo
![image](https://github.com/fpdcc/ccfp-asset-dashboard/assets/114717958/c1c8c14e-f169-4f94-8f5a-ea036201efa9)

### Notes

The funding table scrolls horizontally on small screen sizes. This is the best display I could come up with, but if there is a more appealing way of fitting the table please lmk!

## Testing Instructions
* Create a project and add a phase
  * Confirm that the Update Funding and Add a Source buttons don't render yet while on the creation page 
* After creating a phase, use the Add a source button to add a few rows
* Use the Remove button to get rid of some of the rows
* Test the forms with valid and invalid inputs and hit the Update button
  * Confirm that error messages show appropriately for invalid inputs
  * Confirm that error messages disappear once fixed
* The page should reload after hitting update, when all inputs are valid
  * Confirm that your inputs are saved (indicated by a Delete button next to the fundings)
* Add another a new row, fill in some values, and refresh the page
  * Confirm that a dialog box warns about unsaved changes on the page
